TABCOMPLETION.md

```diff
@@ -0,0 +1,30 @@
+## Tab completion for Fast Downward
+
+We support tab completion for bash and zsh based on the python package [argcomplete](https://pypi.org/project/argcomplete/) which can be installed via apt
+
+```bash
+sudo apt install python3-argcomplete
+```
+
+or pip
+
+```bash
+pip install argcomplete
+```
+
+After the installation, add the following commands to your `.bashrc` or `.zshrc`. Depending on your installation replace `register-python-argcomplete` with `register-python-argcomplete3`.
+
+```
+eval "$(register-python-argcomplete fast-downward.py)"
+eval "$(register-python-argcomplete build.py)"
+eval "$(register-python-argcomplete translate.py)"
+
+function _downward_complete() {
+    local IFS=$'\n'
+    COMPREPLY=( $( "$1" --bash-complete \
+                   "$COMP_POINT" "$COMP_LINE" "$COMP_CWORD" ${COMP_WORDS[@]}))
+}
+complete -o nosort -o default -o bashdefault -F _downward_complete downward
+```
+
+Restart your shell afterwards.
```

Comment:
Do we need to say if/how this affects people that also have other versions of "fast-downward.py" etc. somewhere? Will this break (tab completion for) other programs with the same name? I can imagine that other people have scripts called "build.py", for example, perhaps even "translate.py". If there is a possible interaction, I think it's enough to warn about it and then leave it for people to decide if they care. I don't think we need to be more elaborate (e.g. propose workarounds/solutions).

Reply:
Indeed, the local installation messes with old version or other unrelated files with the same names. The global installation check for a magic comment `# PYTHON_ARGCOMPLETE_OK` and only executes the file if this is found. Unfortunately, the local version does not and simply executes the file when pressing tab. With older versions of build.py it triggers a build. We have the following options:

1. Leave as is and warn about it in the description.
2. Recommend a global activation. This is more intrusive and can cause bugs due to incompatible versions such that sometimes file names are recommended additionally to valid options.
3. Write our own local register function which additionally checks for the magic comment `# PYTHON_ARGCOMPLETE_OK`.

Any thoughts?

Reply:
We discussed this offline and will go with option 1.

Reply:
Now that the argcomplete version that apt installs on Ubuntu 24.04 is compatible with their Python version, we decided to describe both local and global activation. We discuss side effects and potential problems with both variants.